ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -600,36 +622,61 @@ protected void deleteBlob(final String blobIdStr, final String serviceId, Future
      */
     protected void updateBlobTtl(final String blobIdStr, final String serviceId, long expiresAtMs,
         FutureResult<Void> futureResult, Callback<Void> callback) {
-      Callback<GetBlobResultInternal> internalCallback = (GetBlobResultInternal result, Exception exception) -> {
-        if (exception != null) {
-          completeOperation(futureResult, callback, null, exception, false);
-        } else if (result.getBlobResult != null) {
-          exception =
-              new RouterException("GET blob call returned the blob instead of just the store keys (before TTL update)",
-                  RouterErrorCode.UnexpectedInternalError);
-          completeOperation(futureResult, callback, null, exception, false);
-        } else {
-          List<String> blobIdStrs = new ArrayList<>();
-          blobIdStrs.add(blobIdStr);
-          if (result.storeKeys != null) {
-            result.storeKeys.forEach(key -> blobIdStrs.add(key.getID()));
-          }
-          currentOperationsCount.addAndGet(blobIdStrs.size());
-          try {
-            ttlUpdateManager.submitTtlUpdateOperation(blobIdStrs, serviceId, expiresAtMs, futureResult, callback);
-            routerCallback.onPollReady();
-          } catch (RouterException e) {
-            currentOperationsCount.addAndGet(1 - blobIdStrs.size());
-            completeUpdateBlobTtlOperation(e, futureResult, callback);
+
+      // Can skip GET if we can determine this is not a metadata blob
+      boolean needGet = isMaybeMetadataBlob(blobIdStr);
+
+      if (needGet) {
```

Comment:
Is this variable adding any value? I find the following more meaningful:

```java
if (metadata_blob) {
    update_ttl_for_this_and_its_constituents;
  } else {
    update_ttl_only_for_this;
}
```

Reply:
Good point.  Removed.